src/uct/sm/mm/posix/mm_posix.c

```diff
@@ -655,15 +655,14 @@ static uct_mm_md_mapper_ops_t uct_posix_md_ops = {
         .query                  = uct_posix_md_query,
         .mem_alloc              = uct_posix_mem_alloc,
         .mem_free               = uct_posix_mem_free,
-        .mem_advise             = (uct_md_mem_advise_func_t)ucs_empty_function_return_unsupported,
-        .mem_reg                = (uct_md_mem_reg_func_t)ucs_empty_function_return_unsupported,
-        .mem_dereg              = (uct_md_mem_dereg_func_t)ucs_empty_function_return_unsupported,
+        .mem_advise             = ucs_empty_function_return_unsupported,
+        .mem_reg                = ucs_empty_function_return_unsupported,
+        .mem_dereg              = ucs_empty_function_return_unsupported,
         .mkey_pack              = uct_posix_md_mkey_pack,
-        .is_sockaddr_accessible = (uct_md_is_sockaddr_accessible_func_t)ucs_empty_function_return_zero,
-        .detect_memory_type     = (uct_md_detect_memory_type_func_t)ucs_empty_function_return_unsupported
+        .is_sockaddr_accessible = ucs_empty_function_return_zero_int,
```

Comment:
we can't use ucs_empty_function_return_zero here?

Reply:
is_sockaddr_accessible is signed int (...), but ucs_empty_function_return_zero returns unsigned integer and compiler prompts warning

Reply:
ok